packages/neon/neon/lib/src/blocs/user_statuses.dart

```diff
@@ -0,0 +1,87 @@
+part of '../../neon.dart';
+
+abstract class UserStatusesBlocEvents {
+  void load(final String username, {final bool force = false});
+}
+
+abstract class UserStatusesBlocStates {
+  BehaviorSubject<Map<String, Result<NextcloudUserStatusPublicStatus?>>> get statuses;
+}
+
+class UserStatusesBloc extends InteractiveBloc implements UserStatusesBlocEvents, UserStatusesBlocStates {
+  UserStatusesBloc(
+    this._platform,
+    this._account,
+  ) {
+    unawaited(refresh());
+    TimerBloc().registerTimer(const Duration(minutes: 5), refresh);
+  }
+
+  final NeonPlatform _platform;
+  final Account _account;
+  late Timer _timer;
+
+  @override
+  void dispose() {
+    _timer.cancel();
+    unawaited(statuses.close());
+    super.dispose();
+  }
+
+  @override
+  BehaviorSubject<Map<String, Result<NextcloudUserStatusPublicStatus?>>> statuses =
+      BehaviorSubject<Map<String, Result<NextcloudUserStatusPublicStatus?>>>();
+
+  @override
+  Future refresh() async {
+    for (final username in _statuses.keys) {
+      await load(username, force: true);
+    }
+  }
+
+  @override
+  Future load(final String username, {final bool force = false}) async {
+    if (!force && _statuses.containsKey(username)) {
+      return;
+    }
+
+    try {
+      _updateStatus(username, Result.loading());
+
+      late NextcloudUserStatusPublicStatus? data;
+      if (_account.username == username) {
+        final isAway =
+            _platform.canUseWindowManager && (!(await windowManager.isFocused()) || !(await windowManager.isVisible()));
+        data = (await _account.client.userStatus.heartbeat(
+          status: isAway ? NextcloudUserStatusType.away : NextcloudUserStatusType.online,
+        ))
+            .ocs
+            .data
+            .userStatusStatus
+            ?.userStatusPublicStatus;
```

Comment:
I'm not a fan of `(await somefuture).value` I'd rather store the result of await in a separate variable like:

```dart
final results = await _account.client.userStatus.heartbeat(
          status: isAway ? NextcloudUserStatusType.away : NextcloudUserStatusType.online,
        );
data = results.ocs
            .data
            .userStatusStatus
            ?.userStatusPublicStatus;
```